### ISSUE TITLE
Duplicate rows

### ISSUE THREAD

**Issue by SOF3** [poggit/poggit#118]
https://github.com/poggit/poggit/blob/5b7a561bf56367580967d5e199924c6346713b0e/src/poggit/release/index/MainReleaseListPage.php#L68-L74

These JOINs are causing duplicate rows.

**Comment by SOF3** [comment]
Does this still exist?

**Comment by Awzaw** [comment]
That depends if you mean duplicate rows returned by the query, or visible in the UI. The latter is fine, but some tables such as keywords are not normalised which is why we have to eliminate duplicate rows in the loop.

**Comment by SOF3** [comment]
Seems like I'm still getting this in the HTML:
```html
<div id="plugin-apis" value='[["3.0.0-ALPHA4","3.0.0-ALPHA4"],["3.0.0-ALPHA4","3.0.0-ALPHA4"],["2.1.0","2.1.0"],["3.0.0-ALPHA4","3.0.0-ALPHA4"],["2.1.0","2.1.0"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["2.1.0","2.1.0"],["3.0.0-ALPHA4","3.0.0-ALPHA4"],["3.0.0-ALPHA4","3.0.0-ALPHA4"],["2.1.0","2.1.0"],["3.0.0-ALPHA4","3.0.0-ALPHA4"],["2.1.0","2.1.0"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["2.1.0","2.1.0"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["3.0.0-ALPHA4","3.0.0-ALPHA4"],["3.0.0-ALPHA4","3.0.0-ALPHA4"],["2.1.0","2.1.0"],["3.0.0-ALPHA4","3.0.0-ALPHA4"],["2.1.0","2.1.0"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["2.1.0","2.1.0"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["3.0.0-ALPHA4","3.0.0-ALPHA4"],["3.0.0-ALPHA4","3.0.0-ALPHA4"],["2.1.0","2.1.0"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["3.0.0-ALPHA4","3.0.0-ALPHA4"],["3.0.0-ALPHA4","3.0.0-ALPHA4"],["2.1.0","2.1.0"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["3.0.0-ALPHA4","3.0.0-ALPHA4"],["2.1.0","2.1.0"],["3.0.0-ALPHA4","3.0.0-ALPHA4"],["2.1.0","2.1.0"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["2.1.0","2.1.0"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["3.0.0-ALPHA4","3.0.0-ALPHA4"],["2.1.0","2.1.0"],["3.0.0-ALPHA4","3.0.0-ALPHA4"],["2.1.0","2.1.0"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["2.1.0","2.1.0"],["3.0.0-ALPHA4","3.0.0-ALPHA4"],["3.0.0-ALPHA4","3.0.0-ALPHA4"],["2.1.0","2.1.0"],["3.0.0-ALPHA4","3.0.0-ALPHA4"],["2.1.0","2.1.0"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["3.0.0-ALPHA2","3.0.0-ALPHA2"],["2.1.0","2.1.0"]]'></div>
```

This may be bad for the server response in the long run.

**Comment by Awzaw** [comment]
Indeed... I'll check that

**Comment by Awzaw** [comment]
Fixed in ce2fb7c7b
